bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/BoschSHCDeviceHandler.java

```diff
@@ -83,9 +87,44 @@ public void initialize() {
      *         otherwise
      */
     protected boolean processDeviceInfo(Device deviceInfo) {
+        try {
+            updateLocationPropertiesIfApplicable(deviceInfo);
+        } catch (Exception e) {
+            logger.warn("Error while updating location properties for thing {}.", getThing().getUID(), e);
+        }
+        // do not cancel thing initialization if location properties cannot be updated
         return true;
     }
 
+    private void updateLocationPropertiesIfApplicable(Device deviceInfo) throws BoschSHCException {
+        Map<String, String> thingProperties = getThing().getProperties();
+        removeLegacyLocationPropertyIfApplicable(thingProperties);
+        updateLocationPropertyIfApplicable(thingProperties, deviceInfo);
+    }
+
+    private void updateLocationPropertyIfApplicable(Map<String, String> thingProperties, Device deviceInfo)
+            throws BoschSHCException {
+        @Nullable
+        String roomName = getBridgeHandler().resolveRoomId(deviceInfo.roomId);
+        if (roomName != null) {
+            @Nullable
+            String currentLocation = thingProperties.get(PROPERTY_LOCATION);
+            if (currentLocation == null || !currentLocation.equals(roomName)) {
```

Comment:
Can be simplified since `roomName` is already checked as being non-null:
```suggestion
            if (!roomName.equals(currentLocation)) {
```